samples/modules/thrift/hello/server/src/main.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2022 Young Mei
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef __ZEPHYR__
+#include <zephyr/kernel.h>
+#endif
+
+#include <cstdio>
+#include <cstdlib>
+
+#include <thrift/protocol/TBinaryProtocol.h>
+#include <thrift/protocol/TCompactProtocol.h>
+#include <thrift/server/TSimpleServer.h>
+#include <thrift/transport/TBufferTransports.h>
+#include <thrift/transport/TSSLServerSocket.h>
+#include <thrift/transport/TSSLSocket.h>
+#include <thrift/transport/TServerSocket.h>
+
+#include "Hello.h"
+#include "HelloHandler.h"
+
+using namespace ::apache::thrift;
+using namespace ::apache::thrift::protocol;
+using namespace ::apache::thrift::transport;
+using namespace ::apache::thrift::server;
+
+#ifndef IS_ENABLED
+#define IS_ENABLED(flag) flag
+#endif
+
+#ifndef CONFIG_THRIFT_COMPACT_PROTOCOL
+#define CONFIG_THRIFT_COMPACT_PROTOCOL 0
+#endif
+
+#ifndef CONFIG_THRIFT_SSL_SOCKET
+#define CONFIG_THRIFT_SSL_SOCKET 0
+#endif
+
+#ifdef __ZEPHYR__
+int main(void)
+#else
+int main(int argc, char **argv)
+#endif
+{
+	std::string my_addr;
+
+#ifdef __ZEPHYR__
+	my_addr = CONFIG_NET_CONFIG_MY_IPV4_ADDR;
+#else
+	if (IS_ENABLED(CONFIG_THRIFT_SSL_SOCKET)) {
+		if (argc != 5) {
+			printf("usage: %s <ip> <native-cert.pem> <native-key.pem> "
+			       "<qemu-cert.pem>\n",
+			       argv[0]);
+			return EXIT_FAILURE;
+		}
+	} else {
+		if (argc != 2) {
+			printf("usage: %s <ip>\n", argv[0]);
+			return EXIT_FAILURE;
+		}
+	}
+
+	my_addr = std::string(argv[1]);
+#endif
+
+	const int port = 4242;
+	std::shared_ptr<TServerTransport> serverTransport;
+	std::shared_ptr<TTransportFactory> transportFactory;
+	std::shared_ptr<TProtocolFactory> protocolFactory;
+	std::shared_ptr<HelloHandler> handler(new HelloHandler());
+	std::shared_ptr<TProcessor> processor(new HelloProcessor(handler));
+
+	if (IS_ENABLED(CONFIG_THRIFT_SSL_SOCKET)) {
+		std::shared_ptr<TSSLSocketFactory> socketFactory(new TSSLSocketFactory());
+		socketFactory->server(true);
+#ifdef __ZEPHYR__
+		static const char qemu_cert_pem[] = {
+#include "qemu_cert.pem.inc"
```

Comment:
as above - silent failure

Reply:
I see - it should not be a concern for the `Makefile` project, as that is outside the scope of the `#ifdef __ZEPHYR__` block.